src/python/pants/backend/jvm/tasks/nailgun_task.py

```diff
@@ -93,13 +93,16 @@ def create_java_executor(self, dist=None):
       return SubprocessExecutor(dist)
 
   def runjava(self, classpath, main, jvm_options=None, args=None, workunit_name=None,
-              workunit_labels=None, workunit_log_config=None, dist=None):
+              workunit_labels=None, workunit_log_config=None, dist=None, return_workunit=False):
```

Comment:
Same comment here as below: 

Having the return type change based on the value of an optional argument is asking for trouble. How about creating a new method, that returns both, and then calling that from this method, discarding the workunit argument?